mobile/src/main/java/rs/readahead/washington/mobile/media/MediaFileHandler.java

```diff
@@ -742,6 +759,25 @@ public static Single<VaultFile> importVaultFileUri(Context context, @Nullable Ur
         return vaultFile;
     }
 
+    public static List<Single<VaultFile>> importVaultFilesUris(Context context, @Nullable List<Uri> uris, String parentId) throws Exception {
+        List<Single<VaultFile>> vaultFiles = new ArrayList<>();
+        assert uris != null;
+        for (Uri uri : uris) {
+            String mimeType = getMimeType(uri, context.getContentResolver());
+            if (mimeType != null) {
```

Comment:
Let's add a method "importUriByMimeType" that handles the logic for determining which import function to call based on the MIME type.